### ISSUE TITLE
Use operator-sdk run --olm to test operator upgrades

### ISSUE THREAD

**Issue by b1zzu** [operator-framework/operator-sdk#3070]
## Feature Request

**Is your feature request related to a problem? Please describe.**
We would like to test our operator upgrade ( [integreatly-operator](https://github.com/integr8ly/integreatly-operator) ), and to do it we have to install the current stable version of the operator, and then upgrade it to a pre-release version or to the master version. 

What we are currently doing is packaging the stable version and push it to a test repository in quay.io, then we install the operator using OLM form quay, once the installation is finished we build the new version (pre-release or master), push the new operator image and manifests to quay (same test repository) and wait for OLM to automatically trigger the upgrade.

But we would like to remove quay from the process so that it would be also easier to test the upgrade locally, so we tried to use `operator-sdk run --olm` to install a specific version of the operator using OLM without need to push the manifests to quay.io. But we can't reuse the same command to trigger an upgrade because if we run `operator-sdk run --olm` again with a different version it will try to clean up the previously installed version resulting in the uninstall of the operator and then it will install the operator with the new version.

**Describe the solution you'd like**
We would like to have an option in that instead of uninstalling and reinstalling the operator will only update the operator-registry and let OLM detecting the new version and upgrading the operator.

Ex:
```
operator-sdk upgrade --operator-version 2.2.0 --operator-namespace redhat-rhmi-operator --olm --olm-namespace openshift-operator-lifecycle-manager
```

Wdyt? Could this work? Are there better solutions to tests an upgrade?

**Comment by estroz** [comment]
This is actually something we're interested in implementing sooner than later, at least for bundles. It may be reasonable to do both `upgrade bundle` and `upgrade packagemanifests` (`upgrade --olm` in your example) because package manifests support is sticking around for awhile.

/kind feature
/priority important-soon

**Comment by b1zzu** [comment]
Thanks, @estroz, if there is anything else I could help with this feature, let me know :)

**Comment by estroz** [comment]
@b1zzu will do. First we'll need an [enhancement proposal](https://github.com/operator-framework/enhancements), which one of the SDK approvers will be starting sometime in the next few weeks unless another community member takes it up first! Then review and implementation.

**Comment by estroz** [comment]
EP PR for those interested: https://github.com/operator-framework/enhancements/pull/27.

**Comment by estroz** [comment]
Unfortunately this will not make it into v1.0, but will in a soon-to-follow minor release.

**Comment by kensipe** [comment]
Depend on https://github.com/operator-framework/operator-sdk/issues/3626

**Comment by shimron** [comment]
Hi @blzuu , did you find a way to upgrade operator to a newer version ?  I'm also stuck by this step right now. 

https://github.com/operator-framework/operator-sdk/issues/4101

**Comment by rashmigottipati** [comment]
Created this meta-issue https://github.com/operator-framework/operator-sdk/issues/4176 to track run bundle upgrade work.

**Comment by estroz** [comment]
Closed by #4271
